src/pipeline/pipeline.hpp

```diff
@@ -48,11 +54,13 @@ class Pipeline : public Producer<Product>, public Consumer<Consumable> {
     return Nothing();
   }
 
-  //! Ensure that in case of error or empty
-  // result we reset sync consumers to be ready for next iteration.
+  //! Place for additional logic after pipeline run.
+  // Usually we need to ensure that in case of error or empty result we reset
+  // sync consumers to be ready for next iteration. Some of them are crucial
+  // and need to forced to continue the pipeline.
   // Override if you have sync consumers in you pipeline.
   // TODO(bplotka): That would not be needed if we continue pipeline always.
-  virtual Try<Nothing> resetSyncConsumers() {
+  virtual Try<Nothing> postPipelineRun() {
```

Comment:
I have added a note to myself:
"Tagging messages between components with 'author' field could lead to removal of any 'reset' logic in SyncConsumer barrier".
We should get rid of it later, because it rather not conform to simple, elegant and untangled :)